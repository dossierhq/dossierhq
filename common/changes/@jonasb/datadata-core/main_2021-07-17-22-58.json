{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "Add OkResult.map() and ErrorType.isErrorType()",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}