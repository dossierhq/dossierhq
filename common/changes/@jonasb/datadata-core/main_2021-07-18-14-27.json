{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "Configure importHelpers for ts, and add tslib dependency",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}