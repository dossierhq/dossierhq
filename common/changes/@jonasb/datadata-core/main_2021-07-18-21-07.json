{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "Export transpiled JavaScript",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}