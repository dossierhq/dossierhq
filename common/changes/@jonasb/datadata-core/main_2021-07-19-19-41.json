{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "Fix export configuration",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}