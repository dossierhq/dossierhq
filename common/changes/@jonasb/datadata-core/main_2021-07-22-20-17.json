{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "Allow string in schema field specification",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}