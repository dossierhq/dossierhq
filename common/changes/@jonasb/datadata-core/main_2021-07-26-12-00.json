{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "Enable easier JSON serialization of adminClient",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}