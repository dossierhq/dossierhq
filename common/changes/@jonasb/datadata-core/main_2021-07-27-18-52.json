{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "Support boolean type",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}