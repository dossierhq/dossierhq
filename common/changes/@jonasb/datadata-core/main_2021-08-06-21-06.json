{
  "changes": [
    {
      "packageName": "@jonasb/datadata-core",
      "comment": "upsertEntity() support in AdminClient",
      "type": "none"
    }
  ],
  "packageName": "@jonasb/datadata-core",
  "email": "jonas.b@gmail.com"
}