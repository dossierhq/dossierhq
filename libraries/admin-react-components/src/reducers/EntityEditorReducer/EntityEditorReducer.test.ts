import { AdminEntityStatus, AdminSchema } from '@jonasb/datadata-core';
import { Temporal } from '@js-temporal/polyfill';
import type { EntityEditorState, EntityEditorStateAction } from './EntityEditorReducer';
import {
  EntityEditorActions,
  initializeEntityEditorState,
  reduceEntityEditorState,
} from './EntityEditorReducer';

function reduceEntityEditorStateActions(
  state: EntityEditorState,
  ...actions: EntityEditorStateAction[]
) {
  let newState = state;
  for (const action of actions) {
    newState = reduceEntityEditorState(newState, action);
  }
  return newState;
}

describe('initializeEntityEditorState', () => {
  test('no args', () => {
    const state = initializeEntityEditorState();
    expect(state).toMatchInlineSnapshot(`
      Object {
        "activeEntityId": null,
        "drafts": Array [],
        "schema": null,
        "status": "uninitialized",
      }
    `);
  });
});

describe('AddDraftAction', () => {
  test('add Foo', () => {
    const state = reduceEntityEditorState(
      initializeEntityEditorState(),
      new EntityEditorActions.AddDraft({
        id: '619725d7-e583-4544-8bb0-23fc3c2870c0',
        newType: 'Foo',
      })
    );
    expect(state).toMatchSnapshot();
  });
});

describe('SetActiveEntityAction', () => {
  test('add two entities, set active to first', () => {
    const state = reduceEntityEditorStateActions(
      initializeEntityEditorState(),
      new EntityEditorActions.AddDraft({
        id: '619725d7-e583-4544-8bb0-23fc3c2870c0',
        newType: 'Foo',
      }),
      new EntityEditorActions.AddDraft({
        id: '9516465b-935a-4cc7-8b97-ccaca81bbe9a',
        newType: 'Bar',
      }),
      new EntityEditorActions.SetActiveEntity('619725d7-e583-4544-8bb0-23fc3c2870c0')
    );
    expect(state).toMatchSnapshot();
    expect(state.activeEntityId).toBe('619725d7-e583-4544-8bb0-23fc3c2870c0');
  });
});

describe('UpdateEntityAction', () => {
  test('add draft, update entity', () => {
    const id = '619725d7-e583-4544-8bb0-23fc3c2870c0';
    const state = reduceEntityEditorStateActions(
      initializeEntityEditorState(),
      new EntityEditorActions.AddDraft({ id }),
      new EntityEditorActions.UpdateEntity({
        id,
        info: {
          authKey: 'none',
          name: 'Foo name',
          type: 'Foo',
          status: AdminEntityStatus.draft,
          createdAt: Temporal.Instant.from('2022-04-30T07:51:25.56Z'),
          updatedAt: Temporal.Instant.from('2022-04-30T07:51:25.56Z'),
          version: 0,
        },
        fields: {
          title: 'Title',
        },
      })
    );
    expect(state).toMatchSnapshot();
  });
});

describe('UpdateSchemaSpecificationAction', () => {
  test('add empty schema', () => {
    const state = reduceEntityEditorState(
      initializeEntityEditorState(),
      new EntityEditorActions.UpdateSchemaSpecification(
        new AdminSchema({ entityTypes: [], valueTypes: [] })
      )
    );
    expect(state).toMatchSnapshot();
  });
});
