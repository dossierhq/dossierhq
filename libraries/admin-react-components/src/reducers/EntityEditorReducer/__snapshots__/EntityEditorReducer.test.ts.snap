// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDraftAction add Foo 1`] = `
Object {
  "activeEntityEditorScrollSignal": 1,
  "activeEntityId": "619725d7-e583-4544-8bb0-23fc3c2870c0",
  "activeEntityMenuScrollSignal": 1,
  "drafts": Array [
    Object {
      "draft": Object {
        "authKey": null,
        "entitySpec": Object {
          "adminOnly": false,
          "fields": Array [
            Object {
              "name": "title",
              "type": "String",
            },
          ],
          "name": "Foo",
        },
        "fields": Array [
          Object {
            "fieldSpec": Object {
              "name": "title",
              "type": "String",
            },
            "value": null,
          },
        ],
        "name": "",
      },
      "entity": null,
      "id": "619725d7-e583-4544-8bb0-23fc3c2870c0",
    },
  ],
  "schema": AdminSchema {
    "spec": Object {
      "entityTypes": Array [
        Object {
          "adminOnly": false,
          "fields": Array [
            Object {
              "name": "title",
              "type": "String",
            },
          ],
          "name": "Foo",
        },
      ],
      "valueTypes": Array [],
    },
  },
  "status": "uninitialized",
}
`;

exports[`SetActiveEntityAction add two entities, set active to first 1`] = `
Object {
  "activeEntityEditorScrollSignal": 3,
  "activeEntityId": "619725d7-e583-4544-8bb0-23fc3c2870c0",
  "activeEntityMenuScrollSignal": 3,
  "drafts": Array [
    Object {
      "draft": Object {
        "authKey": null,
        "entitySpec": Object {
          "adminOnly": false,
          "fields": Array [],
          "name": "Foo",
        },
        "fields": Array [],
        "name": "",
      },
      "entity": null,
      "id": "619725d7-e583-4544-8bb0-23fc3c2870c0",
    },
    Object {
      "draft": Object {
        "authKey": null,
        "entitySpec": Object {
          "adminOnly": false,
          "fields": Array [],
          "name": "Bar",
        },
        "fields": Array [],
        "name": "",
      },
      "entity": null,
      "id": "9516465b-935a-4cc7-8b97-ccaca81bbe9a",
    },
  ],
  "schema": AdminSchema {
    "spec": Object {
      "entityTypes": Array [
        Object {
          "adminOnly": false,
          "fields": Array [],
          "name": "Foo",
        },
        Object {
          "adminOnly": false,
          "fields": Array [],
          "name": "Bar",
        },
      ],
      "valueTypes": Array [],
    },
  },
  "status": "uninitialized",
}
`;

exports[`SetAuthKeyAction set authKey of new draft 1`] = `
Object {
  "activeEntityEditorScrollSignal": 1,
  "activeEntityId": "619725d7-e583-4544-8bb0-23fc3c2870c0",
  "activeEntityMenuScrollSignal": 1,
  "drafts": Array [
    Object {
      "draft": Object {
        "authKey": "subject",
        "entitySpec": Object {
          "adminOnly": false,
          "fields": Array [
            Object {
              "name": "title",
              "type": "String",
            },
          ],
          "name": "Foo",
        },
        "fields": Array [
          Object {
            "fieldSpec": Object {
              "name": "title",
              "type": "String",
            },
            "value": null,
          },
        ],
        "name": "",
      },
      "entity": null,
      "id": "619725d7-e583-4544-8bb0-23fc3c2870c0",
    },
  ],
  "schema": AdminSchema {
    "spec": Object {
      "entityTypes": Array [
        Object {
          "adminOnly": false,
          "fields": Array [
            Object {
              "name": "title",
              "type": "String",
            },
          ],
          "name": "Foo",
        },
      ],
      "valueTypes": Array [],
    },
  },
  "status": "uninitialized",
}
`;

exports[`SetNameAction set name of new draft 1`] = `
Object {
  "activeEntityEditorScrollSignal": 1,
  "activeEntityId": "619725d7-e583-4544-8bb0-23fc3c2870c0",
  "activeEntityMenuScrollSignal": 1,
  "drafts": Array [
    Object {
      "draft": Object {
        "authKey": null,
        "entitySpec": Object {
          "adminOnly": false,
          "fields": Array [
            Object {
              "name": "title",
              "type": "String",
            },
          ],
          "name": "Foo",
        },
        "fields": Array [
          Object {
            "fieldSpec": Object {
              "name": "title",
              "type": "String",
            },
            "value": null,
          },
        ],
        "name": "New name",
      },
      "entity": null,
      "id": "619725d7-e583-4544-8bb0-23fc3c2870c0",
    },
  ],
  "schema": AdminSchema {
    "spec": Object {
      "entityTypes": Array [
        Object {
          "adminOnly": false,
          "fields": Array [
            Object {
              "name": "title",
              "type": "String",
            },
          ],
          "name": "Foo",
        },
      ],
      "valueTypes": Array [],
    },
  },
  "status": "uninitialized",
}
`;

exports[`UpdateEntityAction add draft, update entity 1`] = `
Object {
  "activeEntityEditorScrollSignal": 1,
  "activeEntityId": "619725d7-e583-4544-8bb0-23fc3c2870c0",
  "activeEntityMenuScrollSignal": 1,
  "drafts": Array [
    Object {
      "draft": Object {
        "authKey": "none",
        "entitySpec": Object {
          "adminOnly": false,
          "fields": Array [
            Object {
              "name": "title",
              "type": "String",
            },
          ],
          "name": "Foo",
        },
        "fields": Array [
          Object {
            "fieldSpec": Object {
              "name": "title",
              "type": "String",
            },
            "value": "Title",
          },
        ],
        "name": "Foo name",
      },
      "entity": Object {
        "fields": Object {
          "title": "Title",
        },
        "id": "619725d7-e583-4544-8bb0-23fc3c2870c0",
        "info": Object {
          "authKey": "none",
          "createdAt": "2022-04-30T07:51:25.56Z",
          "name": "Foo name",
          "status": "draft",
          "type": "Foo",
          "updatedAt": "2022-04-30T07:51:25.56Z",
          "version": 0,
        },
      },
      "id": "619725d7-e583-4544-8bb0-23fc3c2870c0",
    },
  ],
  "schema": AdminSchema {
    "spec": Object {
      "entityTypes": Array [
        Object {
          "adminOnly": false,
          "fields": Array [
            Object {
              "name": "title",
              "type": "String",
            },
          ],
          "name": "Foo",
        },
      ],
      "valueTypes": Array [],
    },
  },
  "status": "uninitialized",
}
`;

exports[`UpdateSchemaSpecificationAction add empty schema 1`] = `
Object {
  "activeEntityEditorScrollSignal": 0,
  "activeEntityId": null,
  "activeEntityMenuScrollSignal": 0,
  "drafts": Array [],
  "schema": AdminSchema {
    "spec": Object {
      "entityTypes": Array [],
      "valueTypes": Array [],
    },
  },
  "status": "uninitialized",
}
`;
